application/data_access/os_api.py

```
@@ -45,24 +54,49 @@ def search_uprn(query: str):
     if results_list is None:
         return []
     return [result.get("DPA", {}) for result in results_list]
 
 
-def search(query: str):
-    type = "uprn" if query.isdigit() else "postcode"
+def search_local_planning_authority(query: str) -> List[Dict]:
+    try:
+        with get_context_session() as session:
+            entity = get_entity_map_lpa(session, {"name": query})
+    except Exception as exc:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)